addons/sale_mrp/models/sale.py

```diff
@@ -15,24 +15,34 @@ class SaleOrder(models.Model):
     @api.depends('procurement_group_id.stock_move_ids.created_production_id.procurement_group_id.mrp_production_ids')
     def _compute_mrp_production_count(self):
         for sale in self:
-            sale.mrp_production_count = len(sale.procurement_group_id.stock_move_ids.created_production_id.procurement_group_id.mrp_production_ids)
+            productions = self.env['mrp.production']
+            moves = sale.procurement_group_id.stock_move_ids
+            for move in moves:
+                while move.move_orig_ids and not move.created_production_id:
+                    move = move.move_orig_ids
+                productions |= move.created_production_id.procurement_group_id.mrp_production_ids
+            sale.mrp_production_count = len(productions)
 
     def action_view_mrp_production(self):
         self.ensure_one()
-        mrp_production_ids = self.procurement_group_id.stock_move_ids.created_production_id.procurement_group_id.mrp_production_ids.ids
+        mrp_productions = self.env['mrp.production']
+        for move in self.procurement_group_id.stock_move_ids:
```

Comment:
Can you create a helper to avoid duplicate code : `_get_mos_generated`

Reply:
Good suggestion, thanks, I've updated the code